src/predis/mock.clj

```diff
@@ -328,6 +341,14 @@
           (replace-seq store k vs')
           v))))
 
+  (core/lset [this k idx v]
+    (let [vs (vec (core/get this k))
+          idx' (normalized-end-idx vs idx)
+          vs' (assoc vs idx' v)]
+      (assert (< idx' (count vs)))
```

Comment:
Should be an error message here matching whatever Redis outputs, see `err-wrongtype` for an example

```clj
(assert condition "error-message")
```